provision.sh

```diff
@@ -14,31 +14,47 @@ echo "---------------------------------------------"
 echo "Running vagrant provisioning"
 echo "---------------------------------------------"
 
+# install heroku toolbelt
+echo "-------------- Installing heroku toolbelt -------------------------"
+#wget -O- https://toolbelt.heroku.com/install-ubuntu.sh | sh
+# These shell script snippets are directly taken from heroku installation script
+# We want to avoid the apt-get update
+# add heroku repository to apt
+echo "deb http://toolbelt.heroku.com/ubuntu ./" > /etc/apt/sources.list.d/heroku.list
+# install heroku's release key for package verification
+wget -O- https://toolbelt.heroku.com/apt/release.key | apt-key add -
+
 echo "------------- Updating packages and their dependencies -----------"
 apt-get update -y # no need for sudo, and -y is needed to bypass the yes-no
+
+# install the toolbelt
+apt-get install heroku-toolbelt --no-install-recommends -y
```

Comment:
It doesn't save in terms of MB. I didn't check, there might be some 1 MB savings or so. But it saves in reduced number of requests to package servers. It saves somewhat in time.